html/janus.js

```diff
@@ -1445,12 +1453,17 @@ function Janus(gatewayCallbacks) {
 						return false;
 					}
 
-					getUserMedia(
-						{audio: audioExist && isAudioSendEnabled(media), video: videoExist ? videoSupport : false},
-						function(stream) { pluginHandle.consentDialog(false); streamsDone(handleId, jsep, media, callbacks, stream); },
-						function(error) { pluginHandle.consentDialog(false); callbacks.error(error); });
+					navigator.mediaDevices.getUserMedia({
+						audio: audioExist ? audioSupport : false,
+						video: videoExist ? videoSupport : false
+					})
+					.then(function(stream) { pluginHandle.consentDialog(false); streamsDone(handleId, jsep, media, callbacks, stream); })
+					.catch(function(error) { pluginHandle.consentDialog(false); callbacks.error(error); });
+				})
```

Comment:
Thanks for the pointer. This seems reasonably implemented around, although I'm aware, for instance, of older Firefox browsers still around in some Janus deployments. That said, it does make sense to try and be more "up-to-date" on the client side too, and older browsers may be configured by web developers to use an older version of janus.js instead.